common.sh

```diff
@@ -112,14 +123,22 @@ function prepare_staging_dir {
 
 # fetch k8s API gen tools and make it available under kb_root_dir/bin.
 function fetch_tools {
-  if [ -n "$SKIP_FETCH_TOOLS" ]; then
-    return 0
+  if [ -z "$SKIP_FETCH_TOOLS" ]; then
+    fetch_kb_tools
+  fi
+}
+
+
+function fetch_go_lint {
+  header_text "Checking for gometalinter.v2"
+  if ! is_installed golangci-lint; then
+    header_text "Installing golangci-lint"
+    curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.21.0
```

Comment:
I'm not a huge fan of `curl | bash`, and this script just downloads a release for the current platform (which we can easily get from `go env GOOS` and `go env GOARCH`).  Can we just do that directly instead?

Reply:
We could get by  using go, however, it is NOT recommended. 
See here: 

<img width="1133" alt="Screenshot 2019-12-05 at 22 02 40" src="https://user-images.githubusercontent.com/7708031/70277816-f9243400-17aa-11ea-8c7a-f7cad7da44b2.png">

And that indeed we already changed in the controller-runtime as well. https://github.com/kubernetes-sigs/controller-runtime/pull/700

Reply:
No, I mean that script just does a whole bunch of nonsense that resolves to 

```
OS=$(go env GOOS)
ARCH=$(go env GOARCH)

curl https://github.com/golangci/golangci-lint/releases/vx.y.z./golang-ci-lint-$OS-ARCH.tar.gz # or something like this
```

We should just do that directly (and fix CR to do that as well).   I've got less of a problem if it's just in CI, but I'd prefer not to be running arbitrary bash scripts from the internet.

Reply:
I understand your concerns. However, since it is the way suggested by golangci-lint, I just moved it for the CI and added a note in the CONTRIBUTION docs

Reply:
Please, let me know if you are ok with.